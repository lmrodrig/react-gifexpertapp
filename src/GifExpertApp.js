import react, {useState} from "react";
import { AddCategory } from "./components/AddCategory";

import { GifGrid } from "./components/GifGrid";

//Funcional Component
const GifExpertApp = () =>{

    const [categories, setCategories] = useState(['One Punch'])
    
    // const handleAdd = () =>{
    //     //forma directa.
    //     //setCategories([...categories,'Yo']);
    //     setCategories( cats => [...cats,'Yo']);
    // }

    return (
        <> 
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr></hr>
            {
            //<button onClick={handleAdd}>Agregar</button>
            }
            
            <ol>
                {
                    categories.map(category =>(
                        <GifGrid 
                            key={category}
                            category= {category}
                        />

                        )
                    )
                    
                    
                }
            </ol>
        </>
    )
}

export default GifExpertApp;