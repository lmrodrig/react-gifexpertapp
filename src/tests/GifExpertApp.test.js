import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Probando GifExpertApp..', () => {
    test('Debe de Mostrar correctamente la applicacion', () => {
      
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de validar el tamaÃ±o de una lista de categorias por default ', () => {
        const categories = ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    
    
});
