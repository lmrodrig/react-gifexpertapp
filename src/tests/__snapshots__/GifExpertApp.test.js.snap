// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifExpertApp.. Debe de Mostrar correctamente la applicacion 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probando GifExpertApp.. Debe de validar el tama√±o de una lista de categorias por default  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
